enhancements/network/on-prem-configurable-lb-dns-stack.md

```diff
@@ -0,0 +1,284 @@
+
+The next table illustrates the supported configuration possibilities:
+
+| Component Name                                       | Current status         |  This proposal          |
+| ---------------------------------------------------- |------------------------| ------------------------|
+| Self-hosted LB and DNS                               |         V              |         V               |
+| Self-hosted LB and DNS(without node names resolution)|         X              |         V               |
+| external LB and DNS                                  |         X              |         V               |
+
+### Goals
+
+- Ability to use the self-hosted stack in all on-prem platforms.
+- Ability to control the self-hosted stack configuration.
+- Retrieve the self-hosted stack status through K8S API.
+- Maintain a single code base/manifests for all the on-prem platforms.
+
+### Non-Goals
+
+- Define a new self-hosted stack for DNS or load balancing
+
+## Proposal
+
+The [Keepalived and infra DNS components](https://github.com/openshift/machine-config-operator/tree/master/manifests/on-prem) on the bootstrap node will keep running through the MCO.
+However, the services on masters and workers nodes will be managed by both MCO and the new operator.
+
+The following functionalities are expected as part of the new operator:
+
+- Introduce a CRD used by the operator to manage the self-hosted stack.
+- Fields in CR spec that control the self-hosted stack configuration.
+- Fields in CR status that expose self-hosted stack status (e.g: which node holds the API-VIP).
+- Run Coredns-MDNS, HAProxy and MDNS-publisher self-hosted stack components as static pods according to the CRD instance content.
+
+## Risks and Mitigations
+
+- Impact on on-prem platforms
+- Upgrades/Downgrades
+
+In order to mitigate the impact on on-prem platforms, in fresh deployments all the self-hosted LB, DNS, and VIP services will be enabled, therefore there is no need to rely on external services during deployment, also the on-prem platforms tests should be updated according to Test Plan section described below.
+
+Additionally, a detailed document in which the supported LB, DNS migration processes will be issued.
+
+## Design Details
+
+### Open Questions [optional]
+
+- Is `cluster-hosted-net-services-operator` an acceptable name?
+- In which namespace the new operator components should be hosted?
+
+### current implmentation
+
+In current implementation, the [networking infra componenets](https://github.com/openshift/installer/blob/master/docs/design/baremetal/networking-infrastructure.md) run as static pods through the MCO.
+
+### Suggested design
+
+To support early clustering requirements Keepalived will continue running as static pods through MCO, additionally, a new dnsmasq service (also deployed through MCO) will run in each node while HAProxy, CoreDNS-MDNS and MDNS-publisher will run as static pods through the new operator.
```

Comment:
This contradicts line 64. Who will run keepalived on masters?

Reply:
Keepalived should run by the MCO and according to the description from line64 services on masters will be managed by both MCO and the new operator.
It seems OK to me.
Am I missing something?